### PR TITLE
add enable / disable / status methods for z coins

### PR DESCRIPTION
**Pull request by smk762**
closes https://github.com/KomodoPlatform/developer-docs/issues/391
- [x] Enabling
- [x] Withdraw
- [x] Sidebar entries
Converted to draft until updated method names are merged in API repo.

### CODE REVIEW

File: docs/basic-docs/atomicdex-api-20-dev/enable_z_coin_init.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+| ----------------------------------------- | --------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| ticker                                    | string          | Ticker of coin to activate                                                                                                                                           |
+| activation_params                         | object          | List of [Electrum servers](https://github.com/KomodoPlatform/coins/tree/master/electrums)                                                                            |
+| activation_params.mode.zcash_params_path  | string          | Optional. Path to folder containing [Zcash parameters](https://z.cash/technology/paramgen/). Defaults to standard location as defined in [this guide](https://forum.komodoplatform.com/t/installing-zcash-params/603) |
```
Comment (laruh) [inline docs/basic-docs/atomicdex-api-20-dev/enable_z_coin_init.md:14]: We should provide a description of `zcash_params_path` after `activation_params.mode`, as its optional parameter and its not a part of `mode`. `zcash_params_path` is one of `activation_params` arguments. 
You can expand the view till activation_params [here](https://codebeautify.org/string-to-json-online/y22489f99)
In addition, you can see, how zcoin activation params look in API code
```rust
pub struct ZcoinActivationParams {
    pub mode: ZcoinRpcMode,
    pub required_confirmations: Option<u64>,
    pub requires_notarization: Option<bool>,
    pub zcash_params_path: Option<String>,
}
```
Where `ZcoinRpcMode`could be `Native` or `Light` with `electrum_servers` and with `light_wallet_d_servers`.

**Review by laruh — COMMENTED** [review]
Looks good!
I have one note

File: docs/basic-docs/atomicdex-api-20-dev/enable_z_coin_init.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+| activation_params.mode.rpc                | string          | Set as `Light` to use external electrum & lightwallet_d servers or `Native` to use local block chain data. If native, the `rpc_data` fields below are not required.  |
+| activation_params.mode.rpc_data           | list of objects | Contains details about servers to be used for `Light` mode operation.                                                                                                |
+| ..rpc_data.electrum_servers.protocol      | string          | Transport protocol used by AtomicDEX API to connect to the electrum server (`TCP` or `SSL`). Optional, defaults to `TCP`                                             |
+| ..rpc_data.electrum_servers.urls          | string          | Urls which are hosting electrum servers                                                                                                                              |
+| ...electrum_servers.disable_cert_verification | boolean     | If `true`, this disables server SSL/TLS certificate verification (e.g. to use self-signed certificate). <b>Use at your own risk</b>                                  |
```
Comment (laruh) [inline docs/basic-docs/atomicdex-api-20-dev/enable_z_coin_init.md:20]: We dont have vector of `urls` in `electrum_servers`. 
We have list of `ElectrumRpcRequest` with parameters: `url`(one string), `protocol` and `disable_cert_verification`. `ElectrumRpcRequest` is structure in our code, we dont use its name in curl, we just use arguments of this structure in curly brackets.

If user wants to use one `url` and put `true` in `disable_cert_verification`, list will look like:
`\"electrum_servers\": [{\"url\":\"zombie.sirseven.me:10033\", \"disable_cert_verification\":true}],`.

Several urls in `electrum_servers` will look like:
`\"electrum_servers\": [{\"url\":\"zombie.sirseven.me:10033\"}, {\"url\":\"zombie.sirseven.me:10033\"}],`.

So basically `electrum_servers` has one parameter, which is a list of `{url, protocol, disable_cert_verification}`

@artemii235 am I right?

**Review by laruh — CHANGES_REQUESTED** [review]
Thanks for the fixes!
Next review iteration :)

**Comment by smk762** [comment]
closes https://github.com/KomodoPlatform/developer-docs/issues/391

File: docs/basic-docs/atomicdex-api-20-dev/enable_z_coin_init.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
+
+## task::enable_z_coin::init
+
+Z coins, like Pirate (ARRR) and the test coin ZOMBIE take a little longer to enable, and use a new two step method to enable. Activation can take a little while the first time, as we need to download some block cache data, and build a wallet database. Subsequent enabling will be faster, but still take a bit longer than other coins. The second step for activation is optional, but allows us to check the status of the activation process.
+<b>To enable Z coins you also need to [install some Zcash Params](https://forum.komodoplatform.com/t/installing-zcash-params/603)</b>
```
[the remark this answers is not in the document]
Comment (smk762) [reply docs/basic-docs/atomicdex-api-20-dev/enable_z_coin_init.md:5]: updated in https://forum.komodoplatform.com/t/installing-zcash-params/603 :+1:

File: docs/basic-docs/atomicdex-api-20-dev/enable_z_coin_init.md (hunk 2 above)
Comment (smk762) [reply docs/basic-docs/atomicdex-api-20-dev/enable_z_coin_init.md]: Updated the table, thanks for pointing it out.

File: docs/basic-docs/atomicdex-api-20-dev/enable_z_coin_status.md [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+```bash
+#!/bin/bash
+source userpass
+curl --url "http://127.0.0.1:7783" --data "
+{
+    \"userpass\": \"${userpass}\",
+    \"method\": \"task::enable_z_coin::status\",
+    \"mmrpc\": \"2.0\",
+    \"params\": {
+        \"task_id\": $1
```
Comment (SirSevenG) [inline docs/basic-docs/atomicdex-api-20-dev/enable_z_coin_status.md:35]: "task_id": $1,
Missing comma here and extra trailing one below

Comment (smk762) [reply docs/basic-docs/atomicdex-api-20-dev/enable_z_coin_status.md]: thanks,fixed.

File: docs/basic-docs/atomicdex-api-20-dev/withdraw_init.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
+# withdraw_init
+
+The `withdraw_init` method generates, signs, and returns a transaction that transfers the `amount` of `coin` to the address indicated in the `to` argument. It is only used for z coins like ZOMBIE, which may take some time to complete. The status of this method can be queried via the [withdraw_status](withdraw_status.html) method, or cancelled with [withdraw_status](withdraw_cancel.html).
```
Comment (laruh) [inline docs/basic-docs/atomicdex-api-20-dev/withdraw_init.md:3]: 1) typo `cancelled with [withdraw_status](withdraw_cancel.html).` - I think this should be `cancelled with [withdraw_cancel](withdraw_cancel.html).` 

2) Could you tell me am I right or wrong? Right now links for `withdraw_status` and  `withdraw_cancel` are invalid. They lead to  our github project pages. But when this branch will be merged into master, links with `withdraw_status.html` and `withdraw_cancel.html` will exist, so everything will be fine.

3) `It is only used for z coins like ZOMBIE` - as I can see in our code the method `task::withdraw::init` is used [not only for `ZCoin`](https://github.com/KomodoPlatform/atomicDEX-API/blob/mm2.1/mm2src/coins/rpc_command/init_withdraw.rs#L115), but also for `UtxoCoin`, `QtumCoin`. In the case of other coins there will be an error `CoinDoesntSupportInitWithdraw`

File: docs/basic-docs/atomicdex-api-20-dev/withdraw_init.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+| coin          | string           | the name of the coin the user desires to withdraw                                                                                         |
+| to            | string           | coins are withdrawn to this address                                                                                                       |
+| amount        | string (numeric) | the amount the user desires to withdraw, ignored when `max=true`                                                                          |
+| max           | bool             | withdraw the maximum available amount                                                                                                     |
+| fee.type      | string           | type of transaction fee; possible values: `UtxoFixed` or `UtxoPerKbyte`                                                                   |
+| fee.amount    | string (numeric) | fee amount in coin units, used only when type is `UtxoFixed` (fixed amount not depending on tx size) or `UtxoPerKbyte` (amount per Kbyte) |
```
Comment (laruh) [inline docs/basic-docs/atomicdex-api-20-dev/withdraw_init.md:15]: We also have arguments `memo` (optional and should be used carefully, as Artem suggested), and [`from`](https://github.com/KomodoPlatform/developer-docs/pull/403/files#r1058952296) (optional). As I can see, we use `from` for `WithdrawRequest` in trezor integration, in the case of Iguana there will be an error `'from' is not supported if the coin is initialized with an Iguana private key`

**Review by laruh — CHANGES_REQUESTED** [review]
Thanks for the fixes!
Next review iteration. I have couple notes and one question :)

File: docs/basic-docs/atomicdex-api-20-dev/withdraw_init.md (hunk 6 above)
Comment (laruh) [reply docs/basic-docs/atomicdex-api-20-dev/withdraw_init.md:15]: > Ain't 100% sure, but `memo` sounds like tendermint-only thing here.

We merged the PR related to memo use in ARRR withdraw https://github.com/KomodoPlatform/atomicDEX-API/pull/1574

File: docs/basic-docs/atomicdex-api-20-dev/withdraw_init.md (hunk 5 above)
Comment (smk762) [reply docs/basic-docs/atomicdex-api-20-dev/withdraw_init.md:3]: Thanks - I've reflowed these docs into two separate docs with subsections, and the sidebar accordingly
image
image
Any z coin specific text was removed from the withdraw tasks doc, with the z coin doc noting that the withdraw tasks are required for those coins and a link to the withdraw task doc.

The links will work once deployed (you can [build & test locally](https://github.com/KomodoPlatform/developer-docs#setup) to confirm. They don't  work in github because the md / html extension change when built.

File: docs/basic-docs/atomicdex-api-20-dev/withdraw_init.md (hunk 6 above)
Comment (smk762) [reply docs/basic-docs/atomicdex-api-20-dev/withdraw_init.md]: Memo tested with ARR successfully (tho not seen in tx_history output as `"error_type":"NotSupportedFor","error_data":"ARRR"`)
Added to the doc along with example to satisfy https://github.com/KomodoPlatform/atomicDEX-API/issues/1589#event-8169743944

Comment (laruh) [reply docs/basic-docs/atomicdex-api-20-dev/withdraw_init.md]: > tho not seen in tx_history output as `"error_type":"NotSupportedFor","error_data":"ARRR"`

Yep, thats correct, `my_tx_history_v2_rpc` now works only for Bch, SlpToken, UtxoCoin, QtumCoin, Tendermint and TendermintToken

File: docs/basic-docs/atomicdex-api-20-dev/withdraw_tasks.md [hunk 7]
```diff
@@ -0,0 +1,297 @@
...
+| coin          | string           | the name of the coin the user desires to withdraw                                                                                         |
+| to            | string           | coins are withdrawn to this address                                                                                                       |
+| amount        | string (numeric) | the amount the user desires to withdraw, ignored when `max=true`                                                                          |
+| memo          | string           | Optional, used for ZHTLC and Tendermint coins only. Attaches a memo to the transaction.                                                   |
+| from          | string           | Optional, used only for transactions using a hardware wallet. For more information, see the [Trezor Integration guide](trezor_integration.html) |
+| max           | bool             | withdraw the maximum available amount                                                                                                     |
```
[the remark this answers is not in the document]
Comment (smk762) [reply docs/basic-docs/atomicdex-api-20-dev/withdraw_tasks.md]: done

File: docs/basic-docs/atomicdex-api-20-dev/withdraw_tasks.md [hunk 8]
```diff
@@ -0,0 +1,297 @@
...
+| fee.type      | string           | type of transaction fee; possible values: `UtxoFixed` or `UtxoPerKbyte`                                                                   |
```
[the remark this answers is not in the document]
Comment (smk762) [reply docs/basic-docs/atomicdex-api-20-dev/withdraw_tasks.md]: done

**Review by laruh — COMMENTED** [review]
Excellent!
Couple more non critical notes

File: docs/basic-docs/atomicdex-api-20/my_tx_history.md [hunk 9]
```diff
@@ -1,6 +1,6 @@
 # my\_tx\_history
 
-To use this method, you must activate your coin with `"tx_history": true`. The response will vary depending on the coin.
+To use this method, you must activate your coin with `"tx_history": true`. The response will vary depending on the coin. Currently only BCH, SLP tokens, UTXO coins, QTUM, and Tendermint/Tendermint tokens are supported.
```
Comment (laruh) [inline docs/basic-docs/atomicdex-api-20/my_tx_history.md:3]: Oh, lets add info that BCH, SLP tokens, UTXO coins, QTUM, and Tendermint/Tendermint are for dev. master branch has less variants: Bch and SlpToken

File: docs/basic-docs/atomicdex-api-20-dev/withdraw_tasks.md [hunk 10]
```diff
@@ -0,0 +1,360 @@
...
+
+
+### Arguments
+
+| Structure     | Type             | Description                                                                                                                               |
+| ------------- | ---------------- | ----------------------------------------------------------------------------------------------------------------------------------------- |
+| coin          | string           | The name of the coin the user desires to withdraw                                                                                         |
+| to            | string           | Coins are withdrawn to this address                                                                                                       |
+| amount        | string (numeric) | The amount the user desires to withdraw, ignored when `max=true`                                                                          |
+| memo          | string           | Optional, used for ZHTLC and Tendermint coins only. Attaches a memo to the transaction.                                                   |
```
Comment (laruh) [inline docs/basic-docs/atomicdex-api-20-dev/withdraw_tasks.md:21]: Please, add @artemii235 note: We advise not to use it to withdraw to CEXes requiring memo.

**Review by SirSevenG — APPROVED** [review]
lgtm

**Review by laruh — APPROVED** [review]
Awesome! :fire:
